coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -32,14 +33,20 @@ def __init__(self, documentation, docstyle_definition,
         :param marker:
             The three-element tuple with marker strings, that identified this
             documentation comment.
-        :param range:
-            The position range of type ``TextRange``.
+        :param position:
+            The starting ``TextPosition`` of the documentation.
         """
         self.documentation = documentation
         self.docstyle_definition = docstyle_definition
         self.indent = indent
         self.marker = marker
-        self.range = range
+        self.position = position
+        self.assemble = None if position is None else self._assemble()
```

Comment:
yup^ although there's a whole revamp coming up in in PHASE-2 when `docbaseclass()` will be implemented. This can be taken care in that^ :wink: .

what i meant by that is the `assemble()` part of `docbaseclass()` we can reset the cache properly there so as to make sure wherever it is called a fresh assembled docstring is fetched. for internal purpose we can use the cached `assemble()` :D